File: backend/app/modules/mutuelle_types/schemas/psc_settings.py
"""Schémas PSC entreprise."""

from __future__ import annotations

from pydantic import BaseModel, Field


class PscSettingsResponse(BaseModel):
    company_id: str
    mutuelle_organisme_label: str | None = None
    mutuelle_employee_self_service: bool = False


class PscSettingsUpdate(BaseModel):
    mutuelle_organisme_label: str | None = Field(None, max_length=128)
    mutuelle_employee_self_service: bool | None = None


class EmployeeMutuelleChoiceRequest(BaseModel):
    mutuelle_type_id: str = Field(..., min_length=1)


class EmployeeMutuelleOption(BaseModel):
    id: str
    libelle: str
    montant_salarial: float
    montant_patronal: float
    pack_couverture: str | None = None
    statut_categoriel: str | None = None
    organisme_label: str | None = None
    organisme_display: str | None = None
    note: str | None = None
    code_option_dsn: str | None = None


class EmployeeMutuelleChoicesResponse(BaseModel):
    organisme_label: str | None = None
    self_service_enabled: bool = False
    current_mutuelle_type_id: str | None = None
    options: list[EmployeeMutuelleOption]

File: frontend/src/api/pscSettings.ts
import apiClient from './apiClient';
import type { MutuelleType } from './mutuelleTypes';

export interface PscSettings {
  company_id: string;
  mutuelle_organisme_label?: string | null;
  mutuelle_employee_self_service: boolean;
}

export interface EmployeeMutuelleOption {
  id: string;
  libelle: string;
  montant_salarial: number;
  montant_patronal: number;
  pack_couverture?: string | null;
  statut_categoriel?: string | null;
  organisme_label?: string | null;
  organisme_display?: string | null;
  note?: string | null;
  code_option_dsn?: string | null;
}

export interface EmployeeMutuelleChoices {
  organisme_label?: string | null;
  self_service_enabled: boolean;
  current_mutuelle_type_id?: string | null;
  options: EmployeeMutuelleOption[];
}

export async function getPscSettings(): Promise<PscSettings> {
  const response = await apiClient.get<PscSettings>('/api/psc-settings');
  return response.data;
}

export async function updatePscSettings(payload: Partial<PscSettings>): Promise<PscSettings> {
  const response = await apiClient.put<PscSettings>('/api/psc-settings', payload);
  return response.data;
}

export async function getAdminPscSettings(companyId: string): Promise<PscSettings> {
  const response = await apiClient.get<PscSettings>(
    `/api/super-admin/companies/${companyId}/psc-settings`,
  );
  return response.data;
}

export async function updateAdminPscSettings(
  companyId: string,
  payload: Partial<PscSettings>,
): Promise<PscSettings> {
  const response = await apiClient.put<PscSettings>(
    `/api/super-admin/companies/${companyId}/psc-settings`,
    payload,
  );
  return response.data;
}

export function pscSettingsClientForCompany(companyId?: string): {
  get: () => Promise<PscSettings>;
  update: (payload: Partial<PscSettings>) => Promise<PscSettings>;
} {
  if (!companyId) {
    return {
      get: getPscSettings,
      update: updatePscSettings,
    };
  }
  return {
    get: () => getAdminPscSettings(companyId),
    update: (payload) => updateAdminPscSettings(companyId, payload),
  };
}

export async function getMyMutuelleChoices(): Promise<EmployeeMutuelleChoices> {
  const response = await apiClient.get<EmployeeMutuelleChoices>('/api/me/mutuelle-choices');
  return response.data;
}

export async function setMyMutuelleChoice(mutuelleTypeId: string): Promise<EmployeeMutuelleChoices> {
  const response = await apiClient.put<EmployeeMutuelleChoices>('/api/me/mutuelle-choice', {
    mutuelle_type_id: mutuelleTypeId,
  });
  return response.data;
}

export type { MutuelleType };
